src/components/interaction-designer/block-types/Core_SetGroupMembershipBlock.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+  <div class="core-set-group-membership-block">
+    <h3 class="no-room-above">
+      {{ 'flow-builder.edit-block-type' | trans({ block_type: trans(`flow-builder.${block.type}`) }) }}
+    </h3>
+
+    <fieldset :disabled="!isEditable">
+      <block-name-editor :block="block"/>
+      <block-label-editor :block="block"/>
+      <block-semantic-label-editor :block="block"/>
+
+      <div class="form-group">
+        <label>{{'flow-builder.action-label' | trans}}</label>
+        <vue-multiselect v-model="selectedAction"
+                         track-by="id"
+                         label="name"
+                         :placeholder="'flow-builder.action-placeholder' | trans"
+                         :options="actionsList"
+                         :allow-empty="true"
+                         :show-labels="false"
+                         :searchable="false">
+        </vue-multiselect>
+      </div>
+
+      <group-selector :block="block"/>
+
+      <first-block-editor-button
+        :flow="flow"
+        :block-id="block.uuid"/>
+
+    </fieldset>
+
+    <block-id :block="block"/>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { namespace } from 'vuex-class'
+import { Component, Prop } from 'vue-property-decorator'
+
+import {
+  IFlow,
+  IBlock,
+  ISetGroupMembershipBlockConfig,
+} from '@floip/flow-runner'
+import BlockNameEditor from '../block-editors/NameEditor.vue'
+import BlockLabelEditor from '../block-editors/LabelEditor.vue'
+import BlockSemanticLabelEditor from '../block-editors/SemanticLabelEditor.vue'
+import FirstBlockEditorButton from '../flow-editors/FirstBlockEditorButton.vue'
+import BlockId from '../block-editors/BlockId.vue'
+import GroupSelector from '@/components/interaction-designer/block-editors/GroupSelector.vue'
+import VueMultiselect from 'vue-multiselect'
+
+import SetGroupMembershipStore, { BLOCK_TYPE, ADD_KEY, REMOVE_KEY } from '@/store/flow/block-types/Core_SetGroupMembershipStore'
+import lang, { trans } from '@/lib/filters/lang'
+import { createDefaultBlockTypeInstallerFor } from '@/store/builder'
+import { find } from 'lodash'
+
+const blockVuexNamespace = namespace(`flow/${BLOCK_TYPE}`)
+const flowVuexNamespace = namespace('flow')
+const builderVuexNamespace = namespace('builder')
+
+interface IGroupOption {
```

Comment:
Can we rename `IGroupOption` to something like `IGroupActionOption` ? 

The difference is then in the shorthand used to reference it which gets misleading. Later we shortened this to `group: IGroupOption `, which then made it into vuex as `group` where the comparison `group.id === ADD_KEY` broke my mind a bit 😹 

If we do the above, can we also rename the variables of that type to be shorthanded as `action` rather than `group`?

Thanks!

Reply:
Very nice catch. I totally agree here. (I made some copy paste from the GroupSelector.vue and forgot we're having `action` not `group` in this component). Renaming ...